spec/shared_examples/user_activation_shared_examples.rb

```diff
@@ -0,0 +1,361 @@
+shared_examples_for 'rails_3_activation_model' do
+  let(:user) { create_new_user }
+  let(:new_user) { build_new_user }
+
+  context 'loaded plugin configuration' do
+    before(:all) do
+      sorcery_reload!([:user_activation], user_activation_mailer: ::SorceryMailer)
+    end
+
+    after(:each) do
+      User.sorcery_config.reset!
+      sorcery_reload!([:user_activation], user_activation_mailer: ::SorceryMailer)
+    end
+
+    it "enables configuration option 'activation_state_attribute_name'" do
+      sorcery_model_property_set(:activation_state_attribute_name, :status)
+
+      expect(User.sorcery_config.activation_state_attribute_name).to eq :status
+    end
+
+    it "enables configuration option 'activation_token_attribute_name'" do
+      sorcery_model_property_set(:activation_token_attribute_name, :code)
+
+      expect(User.sorcery_config.activation_token_attribute_name).to eql :code
+    end
```

Comment:
These will need to be migrated into unit tests, for an example see: https://github.com/Sorcery/sorcery-rework/blob/main/spec/sorcery/plugins/brute_force_protection/model_spec.rb